packages/clerk-js/src/ui/components/UserProfile/EnterpriseAccountsSection.tsx

```diff
@@ -15,58 +31,114 @@ export const EnterpriseAccountsSection = () => {
+  const error = account.verification?.error?.longMessage;
+
+  return (
+    <ProfileSection.Item
+      id='enterpriseAccounts'
+      sx={t => ({
+        gap: t.space.$2,
+        justifyContent: 'start',
+      })}
+      key={account.id}
+    >
+      <EnterpriseAccountProviderIcon account={account} />
+      <Box sx={{ whiteSpace: 'nowrap', overflow: 'hidden' }}>
+        <Flex
+          gap={2}
+          center
+        >
+          <Text
+            truncate
+            colorScheme='body'
+          >
+            {providerName}
+          </Text>
+          <Text
+            truncate
+            as='span'
+            colorScheme='secondary'
+          >
+            {label ? `• ${label}` : ''}
+          </Text>
+          {error && (
+            <Badge
+              colorScheme='danger'
+              localizationKey={localizationKeys('badge__requiresAction')}
+            />
+          )}
+        </Flex>
+      </Box>
+    </ProfileSection.Item>
+  );
+};
+
+const EnterpriseAccountProviderIcon = ({ account }: { account: EnterpriseAccountResource }) => {
+  const { provider, enterpriseConnection } = account;
+  const providerName = getEnterpriseAccountProviderName(account) ?? provider;
+
+  const providerWithoutPrefix = provider.replace('oauth_', '').trim() as OAuthProvider;
+
+  if (isOAuthCustomProvider(provider)) {
+    return enterpriseConnection?.logoPublicUrl ? (
+      <Image
+        elementDescriptor={[descriptors.providerIcon]}
+        elementId={descriptors.providerIcon.setId(providerWithoutPrefix)}
+        alt={providerName}
+        src={enterpriseConnection.logoPublicUrl}
+        sx={theme => ({ width: theme.sizes.$4 })}
+      />
+    ) : (
+      <ProviderInitialIcon
+        id={providerWithoutPrefix}
+        value={providerName}
+        aria-label={`${providerName}'s icon`}
+        elementDescriptor={[descriptors.providerIcon, descriptors.providerInitialIcon]}
+        elementId={descriptors.providerInitialIcon.setId(providerWithoutPrefix)}
+      />
+    );
+  }
+
+  const src = iconImageUrl(isSamlProvider(provider) ? SAML_IDPS[provider].logo : providerWithoutPrefix);
+
+  return (
+    <Image
+      elementDescriptor={[descriptors.providerIcon]}
+      elementId={descriptors.providerIcon.setId(providerWithoutPrefix)}
+      alt={providerName}
+      src={src}
+      sx={theme => ({ width: theme.sizes.$4 })}
+    />
+  );
+};
+
+const getEnterpriseAccountProviderName = ({ provider, enterpriseConnection }: EnterpriseAccountResource) => {
+  if (isSamlProvider(provider)) {
+    return SAML_IDPS[provider]?.name;
+  }
+
+  if (isOAuthCustomProvider(provider)) {
+    return enterpriseConnection?.name;
+  }
+
+  return getOAuthProviderData({ strategy: provider })?.name;
+};
+
+const isSamlProvider = (provider: EnterpriseProvider): provider is SamlIdpSlug => provider.includes('saml');
+
+const isOAuthCustomProvider = (provider: EnterpriseProvider): provider is `oauth_${CustomOauthProvider}` =>
```

Comment:
Ideally, we shouldn't check this against the provider value. It'd be better to check against a unique discriminator property on `EnterpriseAccount`. 

Since we only have `protocol=oauth`, but it could be either EASIE or Enterprise OAuth, then the SDK needs to check against the provider value.

cc @Nikpolik @izaaklauer

Reply:
Agreed if they need different handling they should have a different type.

Any opinions on `oidc` & `easie` VS `oauth_custom` & `oauth_easie`?

Reply:
❓ If this is just for the image, built in OAuth and SAML should have the logo already set. Is there any reason to handle it differently?

Reply:
I've refactored it here to rely on FAPI as the source of truth for connection names, provider names, and logos, instead of mapping to our objects on the SDK. 

The benefit of this is that we won't have to do those explicit checks now against the protocol/provider enum to them access on our own mappings for labels.